include/esp-stub-lib/log.h

```diff
@@ -6,15 +6,58 @@
 
 #pragma once
 
+#if defined(STUB_LOG_ENABLED)
+
+#if !defined(STUB_LIB_LOG_BUF) && !defined(STUB_LIB_LOG_UART)
+#define STUB_LIB_LOG_BUF
+#endif
+
+#if defined(STUB_LIB_LOG_BUF)
 #include <stdint.h>
 
+#if !defined(STUB_LIB_LOG_BUF_SIZE)
+#define STUB_LIB_LOG_BUF_SIZE 4096
+#endif
+
+struct stub_lib_log_buf {
+    uint32_t count;
+    char buf[STUB_LIB_LOG_BUF_SIZE];
+};
+#endif // defined(STUB_LIB_LOG_BUF)
+
 #ifdef __cplusplus
 extern "C" {
 #endif // __cplusplus
 
-void stub_lib_log_init(uint8_t uart_num, uint32_t baudrate);
+void stub_lib_log_init();
 void stub_lib_log_printf(const char *fmt, ...);
 
 #ifdef __cplusplus
 }
 #endif // __cplusplus
+
+#define STUB_LOG_INIT() stub_lib_log_init()
+#define STUB_LOG(fmt, ...) stub_lib_log_printf(fmt, ##__VA_ARGS__)
+#define STUB_LOGE(fmt, ...) STUB_LOG("STUB_E: " fmt, ##__VA_ARGS__)
+#define STUB_LOGW(fmt, ...) STUB_LOG("STUB_W: " fmt, ##__VA_ARGS__)
+#define STUB_LOGI(fmt, ...) STUB_LOG("STUB_I: " fmt, ##__VA_ARGS__)
+#define STUB_LOGD(fmt, ...) STUB_LOG("STUB_D: " fmt, ##__VA_ARGS__)
+#define STUB_LOGV(fmt, ...) STUB_LOG("STUB_V: " fmt, ##__VA_ARGS__)
+// trace only function name
+#define STUB_LOG_TRACE() STUB_LOG("STUB_TRACE %s()\n", __func__)
+// trace with format
+#define STUB_LOG_TRACEF(fmt, ...) STUB_LOG("STUB_TRACE %s(): " fmt, __func__, ##__VA_ARGS__)
+
+#else // defined(STUB_LOG_ENABLED)
+
+#define STUB_LOG_INIT() do {} while(0)
+#define STUB_LOG(fmt, ...) do {} while(0)
+#define STUB_LOGE(fmt, ...) do {} while(0)
+#define STUB_LOGW(fmt, ...) do {} while(0)
+#define STUB_LOGI(fmt, ...) do {} while(0)
+#define STUB_LOGD(fmt, ...) do {} while(0)
+#define STUB_LOGV(fmt, ...) do {} while(0)
```

Comment:
you can define all these `level` logging macros independently of `STUB_LOG_ENABLED` as 
```c
#define STUB_LOGE(fmt, ...) STUB_LOG("STUB_E: " fmt, ##__VA_ARGS__)
#define STUB_LOGW(fmt, ...) STUB_LOG("STUB_W: " fmt, ##__VA_ARGS__)
#define STUB_LOGI(fmt, ...) STUB_LOG("STUB_I: " fmt, ##__VA_ARGS__)
#define STUB_LOGD(fmt, ...) STUB_LOG("STUB_D: " fmt, ##__VA_ARGS__)
#define STUB_LOGV(fmt, ...) STUB_LOG("STUB_V: " fmt, ##__VA_ARGS__)
```
Only `STUB_LOG` will depend on `STUB_LOG_ENABLED`
```c
#if defined(STUB_LIB_LOG_BUF)
#define STUB_LOG(fmt, ...) stub_lib_log_printf(fmt, ##__VA_ARGS__)
#else // defined(STUB_LOG_ENABLED)
#define STUB_LOG(fmt, ...) do {} while(0)
#endif
```
Less code = eaiser to read )))